core/roboconf-target-iaas-vmware/src/main/java/net/roboconf/target/vmware/internal/VmWareMachineConfigurator.java

```diff
@@ -113,7 +114,11 @@ public boolean configure() throws TargetException {
 			return true;
 
 		} catch( Exception e ) {
-			throw new TargetException( e );
+			if(++ this.vmwareToolsCheckCount < 20) {
+				this.logger.fine("VMWare tools not yet started... check #" + this.vmwareToolsCheckCount + " failed");
+				return false;
+			}
+			else throw new TargetException( e );
```

Comment:
I think it is.
If I remove it, I must either return true or false.

Returning true is not an option (it means the config is finished), 
neither returning false (it means that the config is still in progress).

If there is a better way to signal an error than throwing an exception, 
why not... But I don't see which one.


Le 19/02/2016 16:29, Vincent Zurczak a écrit :
>
> In 
> core/roboconf-target-iaas-vmware/src/main/java/net/roboconf/target/vmware/internal/VmWareMachineConfigurator.java 
> <https://github.com/roboconf/roboconf-platform/pull/581#discussion_r53474102>:
>
> > @@ -113,7 +114,11 @@ public boolean configure() throws TargetException {
> >  			return true;
> >
> >  		} catch( Exception e ) {
> > -			throw new TargetException( e );
> > +			if(++ this.vmwareToolsCheckCount < 20) {
> > +				this.logger.fine("VMWare tools not yet started... check #" + this.vmwareToolsCheckCount + " failed");
> > +				return false;
> > +			}
> > +			else throw new TargetException( e );
>
> The "else" is useless here.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/roboconf/roboconf-platform/pull/581/files#r53474102>.
>